mutable_buffer/src/table.rs

```diff
@@ -802,6 +907,151 @@ mod tests {
         );
     }
 
+    #[test]
+    fn write_columns_validates_schema() {
+        let mut dictionary = Dictionary::new();
+        let mut table = Table::new(dictionary.lookup_value_or_insert("foo"));
+
+        let lp = "foo,t1=asdf iv=1i,uv=1u,fv=1.0,bv=true,sv=\"hi\" 1";
+        let entry = lp_to_entry(&lp);
+        table
+            .write_columns(
+                &mut dictionary,
+                0,
+                0,
+                entry
+                    .partition_writes()
+                    .unwrap()
+                    .first()
+                    .unwrap()
+                    .table_batches()
+                    .first()
+                    .unwrap()
+                    .columns(),
+            )
+            .unwrap();
+
+        let lp = "foo t1=\"string\" 1";
+        let entry = lp_to_entry(&lp);
+        let response = table
+            .write_columns(
+                &mut dictionary,
+                0,
+                0,
+                entry
+                    .partition_writes()
+                    .unwrap()
+                    .first()
+                    .unwrap()
+                    .table_batches()
+                    .first()
+                    .unwrap()
+                    .columns(),
+            )
+            .err()
+            .unwrap();
+        assert!(
+            matches!(&response, Error::InternalColumnTypeMismatch {expected_column_type, actual_column_type, ..} if expected_column_type == "tag" && actual_column_type == "String"),
```

Comment:
I don't think `rustfmt` touches text in macros, so it left these long lines alone... I'd split these up a bit though, something like:

```suggestion
            matches!(
                &response, 
                Error::InternalColumnTypeMismatch {
                    expected_column_type, 
                    actual_column_type, 
                    ..
                } if expected_column_type == "tag" && actual_column_type == "String"),
```